src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -408,7 +409,10 @@ func (s *service) FetchTagged(tctx thrift.Context, req *rpc.FetchTaggedRequest)
 	}
 
 	queryResult, err := s.db.QueryIDs(ctx, ns, query, opts)
-	if err != nil {
+	if dberrors.IsUnknownNamespaceError(err) {
+		s.metrics.fetchTagged.ReportError(s.nowFn().Sub(callStart))
+		return nil, convert.ToRPCError(err)
```

Comment:
Hm, wonder if we should just wrap all of our methods that have an `error` type and always just call `convert.ToRPCError(err)` on `err` if it's not nil?

Then it could be like:
```go
if err != nil {
  if _, ok := err.(*rpc.Error); !ok {
    err = convert.ToRPCError(err)
  }
}

return any, other, arg, err
```

However this would require codegen changes unless we started doing this by reflection.

Reply:
Failing that, why don't we just always `convert.ToRPCError(...)` instead of these if/else?  

It will return internal error if the error is not wrapped by an invalid params error, I don't see it necessary to do an if/else here.

Reply:
fair enough, changed to always `convert.ToRPCError`